### PR TITLE
Redirect unavailable Payments > Overview to the Payments > Connect page

### LINKED ISSUE #6011: Payments > Overview not allowed message when force returning from Stripe KYC
### Describe the bug
If a merchant clicks the left-hand side return link on Stripe's KYC, he or she is directed to the **Payments > Overview** page and gets a "Not allowed" message:
Screenshot 2023-04-12 at 17 00 21

### To Reproduce

1. From **WCPay Dev Tools** click the **Re-onboard** link and you will be taken to the Stripe KYC
2. The the **← Return to ...** left-hand side link
3. You will be taken to the **Payments > Overview** page (with a URL like `http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview&wcpay-connection-error=1`) and you will see the error message.

### Expected behavior
Redirect to the Payments > Connect page.

### PR DESCRIPTION
**Pull request by vladolaru**
Fixes #6011 

#### Changes proposed in this Pull Request

When a merchant is directed to the Payments > Overview page (e.g. hitting return in the Stripe KYC) and that WC page is not registered, we redirect them to the Payments > Connect page, if it is registered. This way we avoid a "Not allowed" semi-blank page.

I've also made a small improvement in the JS of the Payments > Overview page to display transient-based error messages also.

#### Testing instructions

* Checkout the PR branch
* Go to WCPay Dev Tools and click on the Re-onboard link
* Once in the Stripe KYC, click the return link
* You should be redirected to the Payment > Connect page.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : QA Testing Not Applicable
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6013` or branch name `fix/6011-payments-overview-redirect` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=fix/6011-payments-overview-redirect) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f462c4f17fce1b304dd7131f461321998c712b77
- Build time: 2023-05-04 06:41:18 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: .gitignore [hunk 1]
```diff
@@ -55,6 +55,16 @@ phpunit.xml
 
 # Logs
 /logs
+*.log
+npm-debug.log*
```
Comment (oaratovskyi) [inline .gitignore]: I guess it wouldn't harm to ignore logs, but maybe data/csv files are needed sometimes under version control, not sure

**Review by oaratovskyi — APPROVED** [review]
Thanks for fixing this one! 🔥 
Looks great to me :shipit:

File: .gitignore (hunk 1 above)
Comment (vladolaru) [reply .gitignore]: Good catch, Oleks. I've taken that list from the server. I will remove these and move them to a separate PR.

**Comment by vladolaru** [comment]
Thank you, @oaratovskyi, for reviewing my PR!

File: .gitignore (hunk 1 above)
Comment (vladolaru) [reply .gitignore]: Handled it in https://github.com/Automattic/woocommerce-payments/pull/6013/commits/3cf9c8b44db492d16c1e2ad8b0e65201b06744c9

**Pull request merged**